sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1768,34 +1772,50 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                 }
             }
 
-            // Some ownership for some previously claimed partitions may have expired or have been stolen; stop the processing for partitions
-            // which are no longer owned.
+            // Some ownership for some previously claimed partitions may have expired or have been stolen; stop the processing
+            // for partitions which are no longer owned.
 
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
-            await Task.WhenAll(ActivePartitionProcessors.Keys
-                .Except(LoadBalancer.OwnedPartitionIds)
-                .Select(partitionId => TryStopProcessingPartitionAsync(partitionId, ProcessingStoppedReason.OwnershipLost, cancellationToken)))
-                .ConfigureAwait(false);
+            foreach (var partitionId in ActivePartitionProcessors.Keys)
+            {
+                if (!LoadBalancer.IsPartitionOwned(partitionId))
+                {
+                   // The partition is no longer owned.  Stopping may take longer than a load balancing cycle
+                   // should run if event processing is active and chooses not to honor cancellation immediately, so
+                   // do not wait for the stop to complete.  Its continuation will ensure proper clean-up even when unobserved.
```

Comment:
Does this mean that even though we don't hold onto the task on L1789, the method will itself will track the async work and it will be awaited when stopping?

Reply:
Correct.  The continuation runs in the background and will be awaited when either the processor is stopped or the same partition is picked up for ownership again.